devito/grid.py

```diff
@@ -0,0 +1,77 @@
+from devito.tools import as_tuple
+from devito.dimension import x, y, z
+
+import numpy as np
+
+__all__ = ['Grid']
+
+
+class Grid(object):
+
+    """
+    A cartesian grid that encapsulates a physical domain over which
+    to discretize :class:`Function`s.
+
+    :param shape: Shape of the domain region in grid points.
+    :param extent: Physical extent of the domain in m; defaults to a
+                   unit box of extent 1m in all dimensions.
+    :param origin: Physical coordinate of the origin of the domain;
+                   defaults to 0. in all dimensions.
+    :param dtype: Default data type to be inherited by all Functions
+                  created from this :class:`Grid`.
+
+    The :class:`Grid` encapsulates the topology and geometry
+    information of the computational domain that :class:`Function`
+    objects can be discretized on. As such it defines and provides the
+    physical coordinate information of the logically cartesian grid
+    underlying the discretized :class:`Function` objects. For example,
+    the conventions for defining the coordinate space in 2D are:
+
+    .. note::
+
+       .. code-block:: python
+
+          x ^
+            |
+            |           origin + extent
+            |     x------------x
+            |     |            |
+            |     |            |
+            |     |   DOMAIN   | extent[1]
+            |     |            |
+            |     |            |
+            |     |  extent[0] |
+            |     x------------x
+            |  origin
+            |
+            |----------------------->
+                       y
```

Comment:
nitpicking
convention is x horizontal, y vertical (matters for angles orientation for tti for example).

Reply:
Ooops, completely true. Will fix.